### PR TITLE
Hack to add Julia 0.6 'primitive', 'struct' and 'mutable' keywords.

### PR DESCRIPTION
**Pull request by m-j-w**
This is a far from perfect hack to enable recognition of Julia 0.6 type keywords `primitive`, `struct` and `mutable`. Perhaps it gives a starting point, or feel free to simply ignore.

### CODE REVIEW

File: src/lexer.jl [hunk 1]
```diff
@@ -954,13 +954,19 @@ function lex_identifier(l, c)
             return tryread(l, ('c', 'r', 'o'), MACRO, c)
         elseif c == 'o'
             return tryread(l, ('d', 'u', 'l', 'e'), MODULE, c)
+        elseif c == 'u'
+            return tryread(l, ('t', 'a', 'b', 'l', 'e'), TYPE, c)
```
Comment (ZacLN) [inline src/lexer.jl]: add new MUTABLE token type to `token_kinds.jl`

File: src/lexer.jl [hunk 2]
```diff
@@ -954,13 +954,19 @@ function lex_identifier(l, c)
...
         else
             return _doret(l, c)
         end
+    elseif c == 'p'
+        return tryread(l, ('r', 'i', 'm', 'i', 't', 'i', 'v', 'e'), BITSTYPE, c)
```
Comment (ZacLN) [inline src/lexer.jl]: new PRIMITIVE token type

File: src/lexer.jl [hunk 3]
```diff
@@ -954,13 +954,19 @@ function lex_identifier(l, c)
...
     elseif c == 'q'
         return tryread(l, ('u', 'o', 't', 'e'), QUOTE, c)
     elseif c == 'r'
         return tryread(l, ('e', 't', 'u', 'r', 'n'), RETURN, c)
+    elseif c == 's'
+        return tryread(l, ('t', 'r', 'u', 'c', 't'), IMMUTABLE, c)
```
Comment (ZacLN) [inline src/lexer.jl]: STRUCT

**Review by ZacLN — COMMENTED** [review]
I think it's worthwhile fully adding these reserved words, including `where`.

It's pretty easy to handle/ignore the new  reserved words downstream and forcing momentum for supporting new versions is good right?

**Comment by m-j-w** [comment]
Added the new token kinds and 'where'. Also added explicit test for combinitation of new keywords.

'PRIMITIVE' is probably a 3-arg keyword ?

**Comment by ZacLN** [comment]
You can ignore those classifications, I'll remove them in another PR

**Comment by ZacLN** [comment]
@KristofferC, thoughts on this?

**Comment by KristofferC** [comment]
Thanks!

**Pull request merged**